packages/flutter/lib/src/material/refresh_indicator.dart

```diff
@@ -319,15 +341,22 @@ class RefreshIndicatorState extends State<RefreshIndicator> with TickerProviderS
       if (_mode == _RefreshIndicatorMode.drag || _mode == _RefreshIndicatorMode.armed)
         _dismiss(_RefreshIndicatorMode.canceled);
     } else if (notification is ScrollUpdateNotification) {
-      if (_mode == _RefreshIndicatorMode.drag || _mode == _RefreshIndicatorMode.armed) {
+      if (notification.dragDetails != null && (_mode == _RefreshIndicatorMode.drag || _mode == _RefreshIndicatorMode.armed)) {
+        final ScrollPosition position = Scrollable.of(notification.context!)!.position;
         if (notification.metrics.extentBefore > 0.0) {
+          // Prevent moving the scroll view until indicator drag offset is fully consumed.
+          position.correctBy(-notification.scrollDelta!);
+        }
+        _dragOffset = _dragOffset! - notification.scrollDelta!;
+        _updateDragOffset(
+          notification.metrics.viewportDimension,
+          // Mitigate an extra friction on iOS.
+          extentPercentage: position.physics is BouncingScrollPhysics ? 0.25 : null,
```

Comment:
This looks like coding specific knowledge of other widgets, do we check physics like this elsewhere in the framework? It might be more appropriate to add an abstract value to ScrollPhysics that this checks for, rather than coding for a specific kind of Scroll Physics.

Reply:
I don't think so. Yeah, it would probably be better as a separate value on ScrollPhysics, I'll see later this week

Reply:
I came up with somethin else - we can use drag delta to actually ignore any friction applied by physics